fix-point/src/CopyFiles.hs

```diff
@@ -0,0 +1,41 @@
+module CopyFiles where
+
+import Control.Applicative
+import Control.Exception (throw)
+import Control.Monad (forM_
+                    , when
+                    , unless)
+import System.Directory (copyFile
+                       , createDirectory
+                       , doesDirectoryExist
+                       , doesFileExist
+                       , listDirectory)
+import System.FilePath ((</>))
+
+import qualified Rename as R
+
+copyAll :: FilePath -> FilePath -> IO ()
+copyAll source target = do
+    unlessM (doesDirectoryExist source) $
+        throw (userError "source does not exist")
+    whenM (doesFileOrDirectoryExist target) $
+        throw (userError "destination already exists")
+
+  -- if bottom two lines swapped, infinite creation of target occurs
+    content <- listDirectory source
+    createDirectory target
+    forM_ content $ \name -> do
+        let sourcePath = source </> name
+        let targetPath = target </> name
+        isDirectory <- doesDirectoryExist sourcePath
+        -- if directory, recurse until a file is reached
+        if isDirectory
+        then copyAll sourcePath targetPath
+        else copyFile sourcePath targetPath
+
+    where
```

Comment:
`where` should be aligned with 2 spaces and functions inside `where` with 4 spaces. So here is the style:

```
foo :: Int -> Int
foo x =
    y + 1
  where
    y = x + x
```

It makes visually more cleaner to separate sections before `where` and after `where`.